contracts/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@rari-capital/solmate/src/utils/ReentrancyGuard.sol";
+import "../refs/CoreRef.sol";
+import "../pcv/IPCVDeposit.sol";
+import "./IPCVSentinel.sol";
+import "./IGuard.sol";
+
+/**
+ * @title PCV Sentinel
+ * @dev the PCV Sentinel should be granted the role Guardian
+ * @notice The PCV Sentinel is a automated extension of the Guardian.
+ * The Guardian can add Guards to the PCV Sentinel. Guards run checks
+ * and provide addresses and calldata for the Sentinel to run, if needed.
+ */
+contract PCVSentinel is IPCVSentinel, CoreRef, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // The set of all guards
+    EnumerableSet.AddressSet private guards;
+
+    /**
+     * @notice Creates a PCV Sentinel with no guards
+     * @param _core the Tribe core address
+     */
+    constructor(address _core) CoreRef(_core) {
+        _setContractAdminRole(keccak256("PCV_SENTINEL_ADMIN_ROLE"));
+    }
+
+    // ---------- Read-Only API ----------
+
+    /**
+     * @notice returns whether or not the given address is a guard
+     * @param guard the address to check
+     */
+    function isGuard(address guard) external view override returns (bool) {
+        return guards.contains(guard);
+    }
+
+    /**
+     * @notice returns a list of all guards
+     */
+    function allGuards() external view override returns (address[] memory all) {
```

Comment:
I think we should add two methods for automation purposes:
1. GetAllProtecableGuards
2. protecAll

so the defender script can run protecAll(getAllProtecableGuards())

Reply:
This can/will be implemented in a helper contract, no need to add it to the core API of the Sentinel

Reply:
(esp because protecAll must necessarily be opinionated about how to handle reverts within single guards when running all of them, etc etc)